### PR TITLE
Fixes disable branding for teams and users

### LINKED ISSUE #5860: "disable branding" only for team plan users
### Is your proposal related to a problem?

currently, any free user can remove the branding, as they are automatically classified as pro users. doesn't it make sense to remove the "disable branding" for free users?

grafik

### Describe the solution you'd like

i think it would be better if only team plan users can do this.

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?
This PR moves branding to only be togglable by people with an active team subscription (Paying customer) 

This just requires the user to be in a active team to get the ability to toggle the option on their own personal events. 

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] To test this login to teampro
- [ ] Toggle you appearance on for personal event types and team event types.
- [ ] Ensure that this reflects the expected settings on your public event type and also the event type that is attached to your team. 
- [ ] Switch to a account that doesnt have any teams and ensure that it gets blocked

### CODE REVIEW

File: apps/web/components/booking/pages/AvailabilityPage.tsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { useAutoAnimate } from "@formkit/auto-animate/react";
 import { EventType } from "@prisma/client";
 import * as Popover from "@radix-ui/react-popover";
+import { useSession } from "next-auth/react";
```
Comment (sean-brydon) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:4]:
```suggestion
```

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 2]
```diff
@@ -134,6 +148,7 @@ const providers: Provider[] = [
         email: user.email,
         name: user.name,
         role: user.role,
+        belongsToActiveTeam: hasActiveTeams,
```
Comment (sean-brydon) [inline apps/web/pages/api/auth/[...nextauth].tsx:151]: Add it to the session

File: apps/web/pages/settings/my-account/appearance.tsx [hunk 3]
```diff
@@ -180,6 +181,7 @@ const AppearanceView = () => {
               <div className="flex-none">
                 <Switch
                   id="hideBranding"
+                  disabled={!session.data?.user.belongsToActiveTeam}
```
Comment (sean-brydon) [inline apps/web/pages/settings/my-account/appearance.tsx:184]: Disable button if we dont have any active teams

File: packages/features/ee/impersonation/lib/ImpersonationProvider.ts [hunk 4]
```diff
@@ -36,6 +37,7 @@ const auditAndReturnNextUser = async (
     name: impersonatedUser.name,
     role: impersonatedUser.role,
     impersonatedByUID,
+    belongsToActiveTeam: hasTeam,
```
Comment (sean-brydon) [inline packages/features/ee/impersonation/lib/ImpersonationProvider.ts:40]: Set this property on impersonation

File: packages/features/ee/impersonation/lib/ImpersonationProvider.ts [hunk 5]
```diff
@@ -103,7 +105,11 @@ const ImpersonationProvider = CredentialsProvider({
       if (impersonatedUser.disableImpersonation) {
         throw new Error("This user has disabled Impersonation.");
       }
-      return auditAndReturnNextUser(impersonatedUser, session?.user.id as number);
+      return auditAndReturnNextUser(
+        impersonatedUser,
+        session?.user.id as number,
+        impersonatedUser.teams.length > 0 // If the user has any teams, they belong to an active team and we can set the hasActiveTeam ctx to true
```
Comment (sean-brydon) [inline packages/features/ee/impersonation/lib/ImpersonationProvider.ts:111]: NIT - ideally this should be checking for active subscriptions - however, this is only used in an admin setting

File: packages/types/next-auth.d.ts [hunk 6]
```diff
@@ -14,6 +14,7 @@ declare module "next-auth" {
     emailVerified?: PrismaUser["emailVerified"];
     email_verified?: boolean;
     impersonatedByUID?: number;
+    belongsToActiveTeam?: boolean;
```
Comment (sean-brydon) [inline packages/types/next-auth.d.ts:17]: Adds property to the session type

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 7]
```diff
@@ -14,12 +14,13 @@ import checkLicense from "@calcom/features/ee/common/server/checkLicense";
 import ImpersonationProvider from "@calcom/features/ee/impersonation/lib/ImpersonationProvider";
 import { hostedCal, isSAMLLoginEnabled } from "@calcom/features/ee/sso/lib/saml";
 import { ErrorCode, isPasswordValid, verifyPassword } from "@calcom/lib/auth";
-import { APP_NAME, WEBAPP_URL } from "@calcom/lib/constants";
+import { APP_NAME, IS_TEAM_BILLING_ENABLED, WEBAPP_URL } from "@calcom/lib/constants";
```
Comment (leog) [inline apps/web/pages/api/auth/[...nextauth].tsx:17]:
```suggestion
import { APP_NAME, WEBAPP_URL } from "@calcom/lib/constants";
```

File: apps/web/pages/api/auth/[...nextauth].tsx [hunk 8]
```diff
@@ -117,6 +123,13 @@ const providers: Provider[] = [
       });
       await limiter.check(10, user.email); // 10 requests per minute
```
Comment (leog) [inline apps/web/pages/api/auth/[...nextauth].tsx:125]:
```suggestion
// Check if the user you are logging into has any active teams
```

File: apps/web/pages/[user]/[type].tsx [hunk 9]
```diff
@@ -150,6 +156,12 @@ async function getUserPageProps(context: GetStaticPropsContext) {
     locations: privacyFilteredLocations(locations),
     descriptionAsSafeHTML: eventType.description ? md.render(eventType.description) : null,
   });
+  const hasActiveTeam =
```
Comment (leog) [inline apps/web/pages/[user]/[type].tsx:159]:
```suggestion
  // Check if the user you are logging into has any active teams
  const hasActiveTeam =
```

**Review by leog — APPROVED** [review]
Some NITs regarding comments and unused variables, LGTM

**Review by PeerRich — APPROVED** [review]

**Pull request merged**